### ISSUE TITLE
Segmentation fault while syncing

### ISSUE THREAD

**Issue by simonerom** [iotexproject/iotex-core#393]
I'm experiencing a new segfault during the sytncig process (on a VPS with 1 vCPU and 2GB Ram - Ubuntu 18.04). Not sure if it's the same issue (#351) I already closed recently, but the the Docker image has not changed (and was never stopped) since then. When I killed the container and restarted a fresh one the node could not sync anymore due to segfault.

The RAM starts filling up around block height 3500, and gets full at around 4500. It seems linked to this log message which pops out more and more frequently until system memory gets filled up: 
```
{"level":"warn","iotxAddr":"io1qyqsqqqq8uhx9jtdc2xp5wx7nxyq3xf4c3jmxknzj23d2m","networkAddress":"137.74.195.242:30555","nodeType":"full_node","time":"2018-12-25T22:26:26Z","message":"dispatcher event chan is full, drop an event"}
```

When the system become almost totally unresponsive, and I try stopping/restarting/killing the container, the "Segmentation fault" message appears on the terminal (system memory is indeed filled up). Once, after I **killed** the container, I got kind of a stack trace I'm attaching here:

[segfault.txt](https://github.com/iotexproject/iotex-core/files/2710577/segfault.txt)

**Comment by simonerom** [comment]
Please find attached the log from the last segfault (this time it happened as soon as  ~ height 130 was reached.
[local.log](https://github.com/iotexproject/iotex-core/files/2714757/local.log)

**Comment by raullenchai** [comment]
Honestly, "1 vCPU and 2GB Ram" is to restricted. Can repro on "2 vCPU and 4GB Ram"? @simonerom

**Comment by simonerom** [comment]
> Honestly, "1 vCPU and 2GB Ram" is to restricted. Can repro on "2 vCPU and 4GB Ram"? @simonerom

I know that! But sometimes making tests on very limited hardware helps finding bugs we don't see on fast machines ;-) Also I wanted to get an idea of how it worked if I run the node on an ARM system: because I could not get the binaries for ARM, running it on a low end configuration was still an interesting try to do! 

Anyway, with the latest releases from v0.4.2 on I have not noticed any segfault anymore. If it happens again, I will make some more tests and let you know.

**Comment by raullenchai** [comment]
Thank @simonerom. Should be fixed in 0.5.0-RC1
